CorsixTH/Lua/app.lua

```diff
@@ -1254,20 +1254,27 @@ function App:checkInstallFolder()
     -- mac_app_dir is the macOS app base directory named CorsixTH.app
     local mac_app_dir = debug.getinfo(1).short_src:match("(.*)/Contents/.")
     local user_dir = os.getenv("HOME") or os.getenv("HOMEPATH") or ""
-    local possible_locations = { user_dir, user_dir .. pathsep ..  "Documents",
+    local possible_locations = {
+      user_dir,
+      user_dir .. pathsep ..  "Documents",
       select(1, corsixth.require("config_finder")):match("(.*[/\\])"):sub(1, -2),
-      mac_app_dir, mac_app_dir and mac_app_dir:match("(.*)/.*%.app"),
+      mac_app_dir or "",
+      mac_app_dir and mac_app_dir:match("(.*)/.*%.app") or "",
       "/Applications/Theme Hospital.app/Contents/Resources/game/Theme Hospital.app/" ..
         "Contents/Resources/Theme Hospital.boxer/C.harddisk",
-      "/usr/share/games/corsix-th", "/usr/local/share/games/corsix-th",
-      os.getenv("ProgramFiles"), os.getenv("ProgramFiles(x86)") }
+      "/usr/share/games/corsix-th",
+      "/usr/local/share/games/corsix-th",
+      os.getenv("ProgramFiles") or "",
+      os.getenv("ProgramFiles(x86)") or "",
+      [[C:]], [[D:]], [[E:]], [[F:]], [[G:]], [[H:]] }
     local possible_folders = { "ThemeHospital", "Theme Hospital", "HOSP", "TH97",
-      [[GOG.com\Theme Hospital]], [[Origin Games\Theme Hospital\data\Game]] }
+      [[GOG.com\Theme Hospital]], [[GOG Games\Theme Hospital]],
+      [[Origin Games\Theme Hospital\data\Game]] }
     for _, dir in ipairs(possible_locations) do
```

Comment:
Couldn't these just be changed to normal pairs if ipairs was finishing prematurely?